### PR TITLE
DAOS-4295: test: Restructuring cont_create for base classes

### PR DESCRIPTION
**Pull request by saurabhtandan**
Re-structuring cont_create() for base classes
of mdtest and fio to use DaosCommand class
implemented in daos_utils.py

Also moving fio_small back to PR run from
weekly run.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

File: src/tests/ftest/io/fio_small.yaml [hunk 1]
```diff
@@ -38,6 +38,8 @@ pool:
   nvme_size: 20000000000
   svcn: 1
   control_method: dmg
+container:
```
Comment (ashleypittman) [inline src/tests/ftest/io/fio_small.yaml]: Is this change related, what does it do?

File: src/tests/ftest/util/mdtest_test_base.py [hunk 2]
```diff
@@ -93,30 +98,30 @@ def _create_pool(self):
...
-        env = Dfuse(self.hostlist_clients, self.tmp).get_default_env()
-        # command to create container of posix type
-        cmd = env + "daos cont create --pool={} --svc={} --type=POSIX".format(
-            self.mdtest_cmd.dfs_pool_uuid.value, self.mdtest_cmd.dfs_svcl.value)
-        try:
-            container = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                         shell=True)
-            (output, err) = container.communicate()
-            self.log.info("Container created with UUID %s", output.split()[3])
+    def _create_cont(self, doas_cmd):
```
Comment (ashleypittman) [inline src/tests/ftest/util/mdtest_test_base.py]: This looks like common code with fio so could be shared but I'm prepared to approve this to move forward.

**Comment by rpadma2** [comment]
Hi Saurabh,
   FYI. I tried to perform this type of change while enabling checksum testing on my PR1943 [ https://github.com/daos-stack/daos/pull/1943 ] and it failed. I tried utilizing both daos API and command lines and it behaved in similar manner.
   Also, I am not sure whether we propagated checksum testing support under 0.9 branch. Some of the checksum test infrastructure might not be there under 0.9. 0.9/1.0 will not have all checksum features enabled.

File: src/tests/ftest/io/fio_small.yaml (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/io/fio_small.yaml]: Yes it is related. This is only specifying a container of type "POSIX". 
Previously we were directly running the command "daos cont create --pool=$DAOS_POOL --svc=$DAOS_SVCL --type=POSIX".
But the output of this command line seems to have got changed a bit recently hence the output which was being caught was different.
So as part of this PR we are doing the same thing but just in a bit better manner using the "daos" utility incorporated in /ftest/utils/daos_util.py

File: src/tests/ftest/util/mdtest_test_base.py (hunk 2 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/mdtest_test_base.py]: I agree. I am only refraining from doing so bcoz I know Ravi is working on enabling aggregation on some of the tests. So I would like to leave it for him to decide.

File: src/tests/ftest/io/fio_small.yaml (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/io/fio_small.yaml]: Ah ok.  If the output of the command truly has changed and is the cause of this then that's a bug with the command isn't it?  Are you able to verify the output directly?

**Comment by ashleypittman** [comment]
> we should then make the same change to mpi4py test if this is now supported

mpi4py is not currently broken, you're correct that if this is the new mechanism then we should update that test to use it but this is an important fix and I don't want to delay it.

**Comment by mchaarawi** [comment]
> > we should then make the same change to mpi4py test if this is now supported
> 
> mpi4py is not currently broken, you're correct that if this is the new mechanism then we should update that test to use it but this is an important fix and I don't want to delay it.

fair enough

**Comment by saurabhtandan** [comment]
Failing tests are due to separate issues
evict_test.py: DAOS-4421
mdtest_small.py: DAOS-4285

**Comment by mchaarawi** [comment]
> LGTM.
> 
> I am still skeptical this will fix fio_small random failures. Based on the discussion with Saurabh, it looks like sometimes for cont_uuid right uuid value was passed in and sometimes different value like "AVOCADO_TEST_BASEDIR" was passed in. This needs to be addressed no matter of the present issue.

can you please point to a PR or a log that has the correct uuid and fails?

**Comment by ashleypittman** [comment]
> > LGTM.
> > I am still skeptical this will fix fio_small random failures. Based on the discussion with Saurabh, it looks like sometimes for cont_uuid right uuid value was passed in and sometimes different value like "AVOCADO_TEST_BASEDIR" was passed in. This needs to be addressed no matter of the present issue.
> 
> can you please point to a PR or a log that has the correct uuid and fails?

I don't think we have one, it's only this PR that has the correct uuid, all others are too old to be still in Jenkins.  Once this lands then we'll starting running this again, and unless the underlying issue has already been fixed then I'd expect us to get back to intermittent failures.

**Comment by ashleypittman** [comment]
> LGTM.
> 
> I am still skeptical this will fix fio_small random failures.

Fixing this issue will take us back to random failures i believe.

> Based on the discussion with Saurabh, it looks like sometimes for cont_uuid right uuid value was passed in and sometimes different value like "AVOCADO_TEST_BASEDIR" was passed in. This needs to be addressed no matter of the present issue.

What happened is we were getting random failures, then the test code regressed so we were getting failures every time, this should resolve the test code issue, which will take us back to this test running what we intend it to.  The intermittent passes all predate the AVOCADO_TEST_BASEDIR failure.

File: src/tests/ftest/util/mdtest_test_base.py (hunk 2 above)
Comment (phender) [reply src/tests/ftest/util/mdtest_test_base.py:101]: I've been working with Amanda to implement a common code approach that would be implemented in the ExecutableCommand class that would define a method that would enable calling an inherited class's 'helper' method - like `container_create()` for DaosCommand - and it would apply a regex to the `CmdResult.stdout` to get the usful information  - in this case the container UUID.
